odc/geo/cog/_tifffile.py

```diff
@@ -0,0 +1,769 @@
+
+
+def _norm_predictor(predictor: Union[int, bool, None], dtype: Any) -> int:
+    if predictor is False or predictor is None:
+        return 1
+
+    if predictor is True:
+        dtype = np.dtype(dtype)
+        if dtype.kind == "f":
+            return 3
+        if dtype.kind in "ui" and dtype.itemsize <= 4:
+            return 2
+        return 1
+    return predictor
+
+
+def _norm_compression_tifffile(
+    dtype: Any,
+    predictor: Union[bool, None, int, Unset] = Unset(),
+    compression: Union[str, Unset] = Unset(),
+    compressionargs: Any = None,
+    level: Optional[Union[int, float]] = None,
+    kw: Optional[Dict[str, Any]] = None,
+) -> Tuple[int, str, Dict[str, Any]]:
+    if kw is None:
+        kw = {}
+    if isinstance(compression, Unset):
+        compression = kw.pop("compress", "ADOBE_DEFLATE")
+        assert isinstance(compression, str)
+
+    if compressionargs is None:
+        compressionargs = {}
+
+    remap = {k.upper(): k for k in kw}
+
+    def opt(name: str, default=None) -> Any:
+        k = remap.get(name.upper(), None)
+        if k is None:
+            return default
+        return kw.pop(k, default)
+
+    def _gdal_level(compression: str, default=None) -> Any:
+        gdal_level_k = GDAL_COMP.get(compression, None)
+        if gdal_level_k is None:
+            return default
+        return opt(gdal_level_k, default)
+
+    compression = compression.upper()
+
+    if level is None and "level" not in compressionargs:
+        # GDAL compat
+        level = _gdal_level(compression)
+
+    if level is not None:
+        compressionargs["level"] = level
+
+    if compression == "DEFLATE":
+        compression = "ADOBE_DEFLATE"
+    if compression == "LERC_DEFLATE":
+        compression = "LERC"
+        compressionargs["compression"] = "deflate"
+        if (lvl := _gdal_level("DEFLATE")) is not None:
+            compressionargs["compressionargs"] = {"level": lvl}
+    elif compression == "LERC_ZSTD":
+        compression = "LERC"
+        compressionargs["compression"] = "zstd"
+        if (lvl := _gdal_level("ZSTD")) is not None:
+            compressionargs["compressionargs"] = {"level": lvl}
+
+    if isinstance(predictor, Unset):
+        predictor = compression in ("ADOBE_DEFLATE", "ZSTD", "LZMA")
+
+    predictor = _norm_predictor(predictor, dtype)
+    return (predictor, compression, compressionargs)
+
+
+def save_cog_with_dask(
+    xx: xr.DataArray,
+    dst: str = "",
+    *,
+    compression: Union[str, Unset] = Unset(),
+    compressionargs: Any = None,
+    level: Optional[Union[int, float]] = None,
+    predictor: Union[int, bool, Unset] = Unset(),
+    blocksize: Union[Unset, int, List[Union[int, Tuple[int, int]]]] = Unset(),
+    bigtiff: bool = True,
+    overview_resampling: Union[int, str] = "nearest",
+    aws: Optional[Dict[str, Any]] = None,
+    client: Any = None,
+    stats: bool | int = True,
+    **kw,
+) -> Any:
+    """
+    Save cloud optimized geotiff to S3 or file.
```

Comment:
Have made a small suggested update below, but a more detailed docstring intro here would be really useful, ideally explaining how this is different from the other COG writing funcs/methods we have (so users can make an informed choice).

```suggestion
    Save a Cloud Optimized GeoTIFF to S3 or file with Dask.
```